### PR TITLE
Add util for comparing (name-mangled) symbols in shared libraries

### PR DESCRIPTION
**Pull request by AljenU**
This util will also be useful whenever NEURON is updated, to check our interface for compatibility

### CODE REVIEW

File: +utils/compare_symbols.m [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+% Use regex to search through full files at once
+for i = 1:nr_symbols
+    %% The symbol in our interface
+    searchstr = list_symbols.Symbol(i);
+    if isequal(searchstr, "hoc_pushs")
+        % terrible workaround
+        % because hoc_pushstr is also in the search list, will find two
+        % symbols matching hoc_pushs, use mangled name as we currently know
+        % it
+        searchstr = "_Z9hoc_pushsP6Symbol";
```
Comment (hvangeffen) [inline +utils/compare_symbols.m:96]: Could possibly match on string followed by capital character or end of string instead

**Review by hvangeffen — COMMENTED** [review]
Worked well when running on my Linux machine as well

File: +utils/compare_symbols.m [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            error("The third part of a line in file starting with MANGLED or NON_MANGLED does not end with ;, script needs to be adapted")
+        end
+        % symbols that are still in neuron_dllimports, but are no longer
+        % used anywhere. Can possibly make this an output of this script,
+        % instead of hardcoding here
+        % if any(contains(["nrn_popsec", "hoc_call_objfunc", "hoc_obj_ref", "hoc_ret", "hoc_objectdata"], parts{3}(1:end-1)))
+        %     continue
+        % end
+        list_symbols.Symbol(i) = parts{3}(1:end-1);
+        list_symbols.is_mangled(i) = line.startsWith('MANGLED');
```
Comment (ramcdougal) [inline +utils/compare_symbols.m:79]: any idea why the first `'` here is being given a dark background?

File: +utils/compare_symbols.m [hunk 3]
```diff
@@ -0,0 +1,149 @@
+function [list_symbols, list_unused] = compare_symbols(optionals)
+%% Compare the symbols that are needed, with the symbols as present in the
```
[the remark this answers is not in the document]
Comment (ramcdougal) [reply +utils/compare_symbols.m:2]: agree; would be nice to clean that up

**Comment by edovanveen** [comment]
TODO before merge:

- [ ] clean up carriage return symbols
- [ ] try Hunter's suggestion for string matching
- [ ] add something to the readme about using the right GCC version

File: +utils/compare_symbols.m (hunk 2 above)
Comment (edovanveen) [reply +utils/compare_symbols.m:79]: VS code does not agree that there's an issue here. Is it a GitLab syntax highlighter quirk? Does it go away if we use double quotes?

File: +utils/compare_symbols.m (hunk 1 above)
Comment (AljenU) [reply +utils/compare_symbols.m]: Even better solution: use negative-lookahead to exclude hoc_pushstr from being found.

**Pull request merged**